cypress-tests/integration-tests/step-definitions/reports-list.ts

```diff
@@ -147,24 +147,28 @@ Then('the column headers are displayed correctly', function (this: Mocha.Context
   const page = new ReportPage()
 
   reportDefinition.variant.specification.fields.forEach((field) => {
-    page.dataTable().find('thead').contains(field.display)
+    if (field.visible) page.dataTable().find('thead').contains(field.display)
   })
 })
 
 Then('date times are displayed in the correct format', function (this: Mocha.Context) {
   const page = new ReportPage()
 
-  reportDefinition.variant.specification.fields.forEach((field, index) => {
-    if (field.type === 'date') {
-      page
-        .dataTable()
-        .get(`tbody tr:first-child td:nth-child(${index + 1})`)
-        .contains(/\d\d\/\d\d\/\d\d \d\d:\d\d/)
-
-      page
-        .dataTable()
-        .get(`tbody tr:nth-child(4) td:nth-child(${index + 1})`)
-        .should('be.empty')
+  let index = 0
```

Comment:
i think you can save a line by using the built-in index: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#callbackfn

Reply:
Yeah the reason of this is because if we use the index of the fields array, it wont correlate to the same index of the column heading if the columns isn't visible

So the intention here is to get the index of the visible column headers

does that make sense?